go/tools/builders/compilepkg.go

```diff
@@ -203,6 +203,7 @@ func compileArchive(
 	}
 	defer cleanup()
 
+	ignoreForNogo := make(map[string]bool)
```

Comment:
Sorry for not spotting this earlier, but is this a length == 1 map?
If that’s the case, could we use a simple `string` var instead?

Reply:
nit for naming: we should probably call this `emptyFiles`, just in case there are other usage of them in the future. If we need to use map as a set, we prefer `map[string]struct{}`, because map[string]bool has 3 states for every entry: true,false,absent, while map[string]struct{} only has two: present,absent, which has less ambiguity.

Reply:
To @sluongng's point, could we avoid needing a set when there is a maximum of one empty file? I think a `string` or `*string` would indeed